assets/app/views/_labels.html

```diff
@@ -0,0 +1,55 @@
+<div ng-controller="LabelsController" class="labels">
+  <div class="gutter-bottom">
+    <div class="flow">
+      <div class="flow-block">
+        <h2>Labels</h2> 
+      </div>
+      <div class="flow-block right">
+        <ul class="list-inline">
+          <li><a class="action action-inline" href ng-click="toggleExpanded()" ng-show="expanded"><i class="pficon pficon-remove"></i>Collapse</a></li>
+          <li><a class="action action-inline" href ng-click="toggleExpanded()" ng-hide="expanded"><i class="pficon pficon-add"></i>Add labels</a></li>
+          <li><span class="help action-inline"><i class="pficon pficon-help"></i> <a href data-toggle="tooltip" data-placement="bottom" title="Labels are a way to filter objects you create in OpenShift.">About labels</a></span></li>
+        </ul>
+      </div>
+    </div>
+    <div class="form-inline labels-edit" ng-show="expanded">
+      <form class="edit-label" name="form">
+        <div ng-class="{'form-group': true, 'has-error': form.labelkey.$error.label}">
+          <label class="sr-only">Label</label>
+          <input class="form-control" type="text" name="labelkey" placeholder="Label" ng-model="labelKey" label-validator required>
+        </div>
+        <span class="separator">:</span>
+        <div class="form-group">
+          <label class="sr-only">Value</label>
+          <input class="form-control" type="text" name="labelvalue" placeholder="Value" ng-model="labelValue" required>
+        </div>
+        <input type="submit" class="btn btn-default" ng-click="addLabel()" ng-enabled="form.$valid" value="Add Label">
+        <div class="has-error" ng-show="form.labelkey.$error.label">
+          <span class="help-block">Please enter a valid object label [[subdomain]/][label]</span>
```

Comment:
i'd pull from https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/identifiers.md#definitions

drop the RFC references and make it a tad more user friendly :)

Maybe do something like "Please enter a valid object label ?" where the ? is a clickable tooltip ( a popover in bootstrap terms ) that has the detailed validation requirements